src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents an contact information.
+ */
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(){
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Ouch! you see that you are comparing parent class with one other child class instance? 
Think how you would alleviate this problem.